src/sync.rs

```diff
@@ -1,6 +1,6 @@
 pub(crate) use self::inner::*;
 
-#[cfg(loom)]
+#[cfg(all(loom, test))]
```

Comment:
This won't work correctly with other crates' tests; `cfg(test)` is only set building tests for _this_ crate. I think this needs to be something like this:
```rust
```suggestion
#[cfg(all(loom, any(test, feature = "loom")))]
```